### PR TITLE
Fix build with clang + small refactoring

### PR DESCRIPTION
**Pull request by amai2012**
subject says it all.

### CODE REVIEW

File: simplecpp.cpp [hunk 1]
```diff
@@ -1737,16 +1737,10 @@ namespace {
         }
     }
 
+    const char * const altopData[] = {"and","or","bitand","bitor","not","not_eq","xor"};
+    const std::set<std::string> altop(&altopData[0], &altopData[7]);
```
Comment (mathbunnyru) [inline simplecpp.cpp]: I think `const std::set<std::string> a{"ab", "cd", ...};` looks much better.

Comment (amai2012) [inline simplecpp.cpp]: That's C++11, isn't it? We need to stick to C++03 here.

Comment (mathbunnyru) [inline simplecpp.cpp]: Yes, that is c++11. Don't know why not to use c++11 or 14 though.

Comment (amai2012) [inline simplecpp.cpp]: Limitation is conservative. It shouldn't have higher requirements than cppcheck (https://github.com/danmar/cppcheck).

Comment (mathbunnyru) [inline simplecpp.cpp]: As I see, cppcheck requires c++11, so is it allowed to use c++11 features in simplecpp?
There are so many beautiful features in c++11.

Comment (danmar) [inline simplecpp.cpp]: Yes C++11 has better features. But this is a library. If somebody wants to use it I want it to be easy to use. If the compiler does not handle C++11 then it is not easy to use.

File: simplecpp.cpp [hunk 2]
```diff
@@ -1737,16 +1739,10 @@ namespace {
         }
     }
 
+    const char * const altopData[] = {"and","or","bitand","bitor","not","not_eq","xor"};
```
Comment (mathbunnyru) [inline simplecpp.cpp]: I'm not sure about codestyle, but usually there is a space after a comma.

Comment (amai2012) [inline simplecpp.cpp]: Maybe this should have been addressed by astyle. I am having problems running it when developing on cygwin. Usually the version is not the required one, or I've seen astyle crashing on cygwin also.

Comment (danmar) [inline simplecpp.cpp]: I don't have a strong opinion about this one. whatever astyle outputs is proper coding style.

File: simplecpp.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/*
+ /*
```
Comment (danmar) [inline simplecpp.cpp]: extra space, typo?

**Pull request merged**